src/averageAges.js

```diff
@@ -55,10 +79,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const women = people.filter(({ sex }) => sex === 'f');
+
+  const children = !onlyWithSon
+    ? people.filter(({ mother }) => mother !== null)
+    : people.filter(({ sex, mother }) => sex === 'm' && mother !== null);
```

Comment:
You have two filter calls - but a cleaner option would be a single filter call with more complex predicate